mathesar/reflection.py

```diff
@@ -39,33 +40,56 @@ def reflect_databases():
 def reflect_schemas_from_database(database):
     engine = create_mathesar_engine(database)
     db_schema_oids = {
-        schema["oid"] for schema in get_mathesar_schemas_with_oids(engine)
+        schema['oid'] for schema in get_mathesar_schemas_with_oids(engine)
     }
 
     database = models.Database.current_objects.get(name=database)
-    schemas = [
+    for oid in db_schema_oids:
         models.Schema.current_objects.get_or_create(oid=oid, database=database)
-        for oid in db_schema_oids
-    ]
     for schema in models.Schema.current_objects.all():
         if schema.database.name == database and schema.oid not in db_schema_oids:
             schema.delete()
-    return schemas
 
 
 def reflect_tables_from_schema(schema):
     db_table_oids = {
-        table["oid"]
+        table['oid']
         for table in get_table_oids_from_schema(schema.oid, schema._sa_engine)
     }
-    tables = [
+    for oid in db_table_oids:
         models.Table.current_objects.get_or_create(oid=oid, schema=schema)
-        for oid in db_table_oids
-    ]
-    for table in models.Table.current_objects.all().filter(schema=schema):
+    for table in models.Table.current_objects.filter(schema=schema):
         if table.oid not in db_table_oids:
             table.delete()
-    return tables
+
+
+def reflect_constraints_from_database(database):
+    engine = create_mathesar_engine(database)
+    db_constraints = get_constraints_with_oids(engine)
+    for db_constraint in db_constraints:
+        try:
+            models.Constraint.current_objects.get_or_create(
+                oid=db_constraint['oid'],
+                table=models.Table.current_objects.get(oid=db_constraint['conrelid'])
+            )
+        except models.Table.DoesNotExist:
+            pass
```

Comment:
I agree the readability could be improved. Let me know if the version I pushed looks better. There really shouldn't be duplicate OIDs, given that it's the primary identifier of the table in the database, so I deliberately didn't handle that case, I'd rather it throw an error.

Reply:
The OIDs do roll over at some point, but that would be an _extreme_ edge case, since the user would have needed to create many _many_ more tables than the throughput of our app is probably going to enable in a reasonable timeframe.  I definitely think throwing an error would appropriate.  An amusing (if ancient) thread from the PostgreSQL mailing list:

https://www.postgresql.org/message-id/Pine.GSO.4.02A.10004101348060.8269-100000%40Falk.DoCS.UU.SE

Reply:
@kgodey looks great, thank you!